### DISCUSSION TITLE
Auto bullets and Auto next lvl

### DISCUSSION THREAD

**Discussion by allreddv** [MeanderingProgrammer/render-markdown.nvim#219] [Ideas]
First, not sure if you know this, but if you go to the neovim.io and click plugins, render-markdown is the #1 trending plugin at the moment, so congrats.  

I have bullets working exactly how I like them now thanks to your feature add, and I also see you added ability to use icons for ordered lists.  I just thought I would throw this out there as a possible feature upgrade since you seem to be actively  doing a lot of development still.  

Right now i have things working by combining render-markdown with bullets.vim, well actually by using kaymmm / bullets.nvim which is a lua port for bullets.vim.  I couldn't get it to work correctly with bullets.vim but could with the lua port for some reason..  I just use all the auto functionality and have the bullets type list in bullets.nvim just set as an empty list so then it applies its auto features to the render-markdown bullets I set up with the list of lists. 

This combination makes it really nice to organize note taking into bullet lists by automatically placing the next bullet when you press enter and if you want to go to the next level down you just end the last line of the lvl you are in with a colon and it will add the next level of bullet on the next line and continues on.  Also if you hit enter on the last line without entering text it will delete the bullet it added and take you back a lvl.  

It also adds the ability to promote or demote a bullet's lvl  and then automatically renumbers everything in the list accordingly. which is really useful.  

Anyways, like I said, I have it working this way with a combination of both plugins, but it could be a nice feature to have the functionality built into render-markdown. 

Why not just use bullets.nvim for bullets?   Because tree-sitter won't recognize its structure as a list and list items, but onlyt as a paragraph so you can't get any highlighting at all.  I can even get treesitter to identify the items as list-items using a custom query which I can then see using tree-sitters playground, but even then, InspectTree still sees them as only paragraphs and line items in the paragraph.  I think it has something to do with the markdown language server itself.  Also,  I don't think the plugin is getting any current maintenance or anything, so it could break anytime depending on other updates. 

I will try to post a small clip of the functionality working together today if I get a chance.

**Comment by MeanderingProgrammer** [comment]
Thank you :) The plugin does sometimes make it to the top of the trending page and pretty regularly stays on the first page. It's really awesome that a relatively large chunk of this community likes the plugin, I definitely didn't expect it but I do appreciate it!

I operate under the mental model that this plugin is rendering only. It doesn't modify the underlying text in any way and it provides no keymaps / cursor hijacking behaviors. This keeps the scope reasonable, as opposed to being a one stop plugin for all your markdown needs, which is not a goal I have for it.
 
I think other plugins like [markdown.nvim](https://github.com/tadmccorkle/markdown.nvim) do an awesome job adding helpful keymaps for interacting with markdown, LSPs add useful linking information, and formatters help keep the text well structured.

I also think opting in to these features individually by adding more plugins is a good way to keep things structured both for users and maintainers.

I appreciate the suggestion and if you have any others around rendering behaviors definitely let me know!

It is very strange that `bullets.nvim` would in any way impact treesitters ability to understand lists / list items, I don't think that's intended behavior.
